src/controller/python/chip-device-ctrl.py

```diff
@@ -516,10 +516,33 @@ def do_resolve(self, line):
             print(str(ex))
             return
 
+    def wait_for_one_discovered_device(self):
+        print("Waiting for device responses...")
+        strlen = 100;
+        addrStrStorage = ctypes.create_string_buffer(strlen)
+        count = 0
+        maxWaitTime = 2
+        while (not self.devCtrl.GetIPForDiscoveredDevice(0, addrStrStorage, strlen) and count < maxWaitTime):
+            time.sleep(0.2)
+            count = count + 0.2
+
+    def wait_for_many_discovered_devices(self):
+        # Discovery happens through mdns, which means we need to wait for responses to come back.
+        # TODO(cecille): I suppose we could make this a command line arg. Or Add a callback when
+        # x number of responses are received. For now, just 2 seconds. We can all wait that long.
+        print("Waiting for device responses...")
+        time.sleep(2)
+
     def do_discover(self, line):
```

Comment:
Although not merged, https://github.com/project-chip/connectedhomeip/pull/7432 has introduced use of argparse to remove the need for all this boilerplate just to define the args interface.

This PR is expanding the number of arguments for do_discover from 2 to 8, so is nearly a complete refactor.  Given this, would you consider using argparse as well here?  It's just a whole lot easier.

Reply:
Changed over in next commit. I'm not going to tackle any other refactors in this PR. I suppose this is ok - it's a bit longer, but does essentially the same thing.

Reply:
Thanks!